xbmc/music/MusicDatabase.cpp

```diff
@@ -8888,6 +8942,184 @@ bool CMusicDatabase::GetGenresJSON(CFileItemList& items, bool bSources)
   return false;
 }
 
+bool CMusicDatabase::GetBoxsetsAlbums(const std::string& strBaseDir, CFileItemList& items)
+{
+  CMusicDbUrl musicUrl;
+
+  if (!musicUrl.FromString(strBaseDir))
+    return false;
+
+  musicUrl.AddOption("boxset", true);
+
+  Filter filter;
+  return GetAlbumsByWhere(musicUrl.ToString(), filter, items);
+}
+
+bool CMusicDatabase::GetBoxsetDiscs(const std::string& strBaseDir, CFileItemList& items, int idAlbum, const SortDescription &sortDescription)
+{
+  // Need to get the single album - must only be one album holding a boxset
+  // then split the disc names from strDiscTitles, count how many there are
+  // and loop around that many times to create a view with 'i' number of boxes,
+  // titled by the subtitles
+
```

Comment:
Note to myself (because it may not be clear)
Check album first, then get disc tiles and number - both held in item along with album data. Use both idAlbum and disc number in path to use as index  to get then songs for that disc